cpp/libs/ossrf_gstreamer_api/lib/src/receiver/st2110_30_receiver_plugin.cpp

```diff
@@ -0,0 +1,110 @@
+    {
+        // Create pipeline and check if all elements are created successfully
+        BST_CHECK_ASSIGN(pipeline_, bisect::gst::pipeline::create(NULL));
+        auto* pipeline = pipeline_.get();
+
+        // Add pipeline udp source
+        auto* source = gst_element_factory_make("udpsrc", NULL);
+        BST_ENFORCE(source != nullptr, "Failed creating GStreamer element udpsrc");
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), source), "Failed adding udpsrc to the pipeline");
+
+        // Set udp source params
+        g_object_set(G_OBJECT(source), "address", s_.primary.source_ip_address.c_str(), NULL);
+        g_object_set(G_OBJECT(source), "auto-multicast", TRUE, NULL);
+        g_object_set(G_OBJECT(source), "port", s_.primary.source_port, NULL);
+        g_object_set(G_OBJECT(source), "multicast-iface", s_.primary.interface_name.c_str(), NULL);
+
+        // Create and set caps for udp source
+        GstCaps* caps = gst_caps_from_string("application/x-rtp, clock-rate=48000, channels=2");
+        g_object_set(G_OBJECT(source), "caps", caps, NULL);
+
+        // Add pipeline rtpjitterbuffer
+        auto* jitter_buffer = gst_element_factory_make("rtpjitterbuffer", NULL);
+        BST_ENFORCE(jitter_buffer != nullptr, "Failed creating GStreamer element jitter_buffer");
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), jitter_buffer), "Failed adding jitter_buffer to the pipeline");
+
+        // Add pipeline queue1
+        auto* queue1 = gst_element_factory_make("queue", NULL);
+        BST_ENFORCE(queue1 != nullptr, "Failed creating GStreamer element queue");
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), queue1), "Failed adding queue to the pipeline");
+        g_object_set(G_OBJECT(queue1), "max-size-time", queue_max_size_time, "max-size-buffers", queue_max_size_buffers,
+                     "max-size-bytes", queue_max_size_bytes, NULL);
+
+        // Add pipeline rtp depay
+        auto* depay = gst_element_factory_make("rtpL24depay", NULL);
+        BST_ENFORCE(depay != nullptr, "Failed creating GStreamer element depay");
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), depay), "Failed adding depay to the pipeline");
+
+        // Add pipeline video sink
+        auto* sink = gst_element_factory_make("pulsesink", NULL);
+        BST_ENFORCE(sink != nullptr, "Failed creating GStreamer element sink");
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), sink), "Failed adding sink to the pipeline");
+
+        // Link all elements together
+        BST_ENFORCE(gst_element_link_many(source, jitter_buffer, queue1, depay, sink, NULL),
+                    "Failed linking GStreamer video pipeline");
+
+        // Setup runner
+        pipeline_.run_loop();
+
+        return {};
+    }
+
+    void stop() noexcept override
+    {
+        pipeline_.stop();
+        pipeline_ = {};
+    }
+};
+
+// TODO this function will need to receive an SDP and use the information in it to build the GST pipeline
```

Comment:
This function already receives information from the SDP, so this todo seams outdated